### PR TITLE
Collapse pipeline aggs into single package

### PR DESCRIPTION
**Pull request by polyfractal**
Collapses all the `org.elasticsearch.search.aggregations.pipeline.*` packages down to a single level.

- Restrict visibility of Aggregators and Factories
- Move PipelineAggregatorBuilders up a level so it is consistent with AggregatorBuilders
- Checkstyle line length fixes for a few classes
- Minor odds/ends (swapping to method references, formatting, private variables, deprecated logger, etc)

Relates #22868

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketSelectorPipelineAggregator.java [hunk 1]
```diff
@@ -40,14 +38,12 @@
...
-
     private Script script;
-
     private Map<String, String> bucketsPathsMap;
 
-    public BucketSelectorPipelineAggregator(String name, Map<String, String> bucketsPathsMap, Script script, GapPolicy gapPolicy,
+    BucketSelectorPipelineAggregator(String name, Map<String, String> bucketsPathsMap, Script script, GapPolicy gapPolicy,
             Map<String, Object> metadata) {
-        super(name, bucketsPathsMap.values().toArray(new String[bucketsPathsMap.size()]), metadata);
+        super(name, bucketsPathsMap.values().toArray(new String[0]), metadata);
```
Comment (colings86) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketSelectorPipelineAggregator.java]: I'm curious why this change is needed? My understanding was that if you initialise the array to the size of the map it avoids having to perform resizes which reinitialise the array inside the `toArray()` method.

File: server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketScriptPipelineAggregator.java [hunk 2]
```diff
@@ -48,9 +46,9 @@
     private final Script script;
     private final Map<String, String> bucketsPathsMap;
 
-    public BucketScriptPipelineAggregator(String name, Map<String, String> bucketsPathsMap, Script script, DocValueFormat formatter,
+    BucketScriptPipelineAggregator(String name, Map<String, String> bucketsPathsMap, Script script, DocValueFormat formatter,
             GapPolicy gapPolicy, Map<String, Object> metadata) {
-        super(name, bucketsPathsMap.values().toArray(new String[bucketsPathsMap.size()]), metadata);
+        super(name, bucketsPathsMap.values().toArray(new String[0]), metadata);
```
Comment (colings86) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketScriptPipelineAggregator.java]: I'm curious why this change is needed? My understanding was that if you initialise the array to the size of the map it avoids having to perform resizes which reinitialise the array inside the `toArray()` method.

Comment (colings86) [inline server/src/main/java/org/elasticsearch/search/aggregations/pipeline/BucketScriptPipelineAggregator.java]: I'm curious why this change is needed? My understanding was that if you initialise the array to the size of the map it avoids having to perform resizes which reinitialise the array inside the `toArray()` method.

**Review by colings86 — APPROVED** [review]
LGTM - I left a small comment more out of curiosity than anything else

**Review by colings86 — APPROVED** [review]
LGTM

**Review by jimczi — APPROVED** [review]
LGTM

**Review by jimczi — APPROVED** [review]
LGTM
